EuclideanGeometry/Example/ShanZun/Ex1.lean

```diff
@@ -14,9 +14,27 @@ namespace Shan_Problem_1_1
 /- In $\triangle ABC$, let $AD$ be the height and $AE$ be the angle bisector of $\triangle ABC$.
 Prove that $\angle DAE = (\angle CBA - \angle ACB) / 2$. -/
 
+-- We have triangle $\triangle ABC$
+variable {A B C : P} {hnd : ¬ colinear A B C}
+-- Claim: $A \ne B$ and $B \ne C$ and $C \ne A$.
+lemma a_ne_b : A ≠ B := sorry
+lemma b_ne_c : B ≠ C := sorry
+lemma c_ne_a : C ≠ A := sorry
+-- $AD$ is the height --`May change due to construction of height in future`
+variable {D : P} {hd : D = perp_foot A (LIN B C b_ne_c.symm)}
+-- $AE$ be the angle bisector of $\triangle ABC$ --`May change in future, a triangle_nd's AngBis should be a Seg_nd`
+variable {E : P} {he : is_inx E (SEG B C) (ANG A B C a_ne_b b_ne_c.symm).AngBis}
+-- Claim : $D \ne A$ and $E \ne A$
+lemma d_ne_a : D ≠ A := sorry
+lemma e_ne_a : E ≠ A := sorry
 
--- Let $\triangle ABC$ be an nondegenerate triangle.
-
+theorem Shan_Problem_1_1 : ∠ D A E d_ne_a e_ne_A = (1 / 2) * (∠ C B A b_ne_c.symm a_ne_b - ∠ A C B c_ne_a.symm b_ne_c) := by
+  have ang₁ : ∠ B A E a_ne_b.symm e_ne_a - ∠ D A E d_ne_a e_ne_a = ∠ B A D a_ne_b d_ne_a := sorry
+  have ang₂ : ∠ E A C e_ne_a c_ne_a + ∠ D A E d_ne_a e_ne_a = ∠ D A C d_ne_a c_ne_a := sorry
+  have ang₃ : ∠ B A E a_ne_b.symm e_ne_a = ∠ E A C e_ne_a c_ne_a := sorry
+  have ang₄ : ∠ D A C d_ne_a c_ne_a - ∠ B A D a_ne_b d_ne_a = ∠ C B A b_ne_c.symm a_ne_b - ∠ A C B c_ne_a.symm b_ne_c := sorry
+  rw[← ang₄, ← ang₂, ← ang₁, ← ang₃]
+  ring
 end Shan_Problem_1_1
 
 theorem Seg_nd.not_lies_int_extn_and_rev_extn_of_lies_on {A : P} {seg_nd : Seg_nd P} (lieson : A LiesOn seg_nd.1) : ¬ A LiesInt seg_nd.extension ∧ ¬ A LiesInt seg_nd.reverse.extension := by

```

Comment:
Please create a ray.trash file and put this lemma into it. Then import this file.